azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py

```diff
@@ -119,3 +136,58 @@ def ingest_from_blob(self, blob_descriptor, ingestion_properties):
         ingestion_blob_info_json = ingestion_blob_info.to_json()
         encoded = base64.b64encode(ingestion_blob_info_json.encode("utf-8")).decode("utf-8")
         queue_service.put_message(queue_name=queue_details.object_name, content=encoded)
+
+    def ingest_from_stream(self, stream, ingestion_properties):
+        """Ingest from in-memroy streams.
+        :param BytesIO or StringIO stream: in-memory stream.
+        :param azure.kusto.ingest.IngestionProperties ingestion_properties: Ingestion properties.
+        """
+
+        if isinstance(stream, io.BytesIO) or isinstance(stream, io.StringIO):
+            stream.seek(0, io.SEEK_END)
```

Comment:
isn't this somewhat expensive? can we assume give a stream we are always at the end of it? or at least check for that?